server/status.go

```diff
@@ -223,6 +236,66 @@ func (s *statusServer) handleLocalLog(w http.ResponseWriter, r *http.Request, ps
 	w.Write(b)
 }
 
+// handleLocalLog returns the log entires parsed from the log file that occurred
+// after or on the query parameter "starttime" and before or on the query
+// parameter "endtime". Additionally, if a level is specified via the URL, the
+// log entires are scoped to that level. If no "starttime" is provided, then
+// a "starttime" of a day ago is used.  If no "endtime" is provided, the current
+// time is used. The cutoff for entires is defined in log.EntriesCutoff.
+func (s *statusServer) handleLocalLog(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	log.Flush()
+	level, _ := log.LevelFromString(ps.ByName("level"))
+	startTime := r.URL.Query().Get("starttime")
+	var startTimeNano int64
+	if len(startTime) > 0 {
+		var err error
+		startTimeNano, err = strconv.ParseInt(startTime, 10, 0)
+		if err != nil {
+			log.Error(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+	} else {
+		startTimeNano = time.Now().AddDate(0, 0, -1).UnixNano()
+	}
+
+	endTime := r.URL.Query().Get("endtime")
```

Comment:
Done.  Not sure how much this saves, as it still needs to do the same error checking.  But it goes look a bit nicer.